src/transformations/transform-constructor.ts

```diff
@@ -78,86 +81,110 @@ export function* removeLeftoverConstructorHead(sourceUnit: SourceUnit): Generato
-
-    const initializer = (
-      helper: TransformHelper,
-      argsList = '',
-      unchainedArgsList = '',
-      argNames: string[] = [],
-    ) => [
-      `function __${name}_init(${argsList}) internal onlyInitializing {`,
-      buildSuperCallsForChain(contractNode, tools, helper),
-      emptyConstructor ? [] : [`__${name}_init_unchained(${argNames.join(', ')});`],
-      `}`,
-      ``,
-      [
-        `function`,
-        `__${name}_init_unchained(${unchainedArgsList})`,
-        `internal onlyInitializing`,
-        ...modifiers.map(m => helper.read(m)),
-        `{`,
-      ].join(' '),
-      varInitNodes.map(v => {
-        const newExpr = parseNewExpression(v.value!);
-        if (!newExpr) {
-          return `${v.name} = ${helper.read(v.value!)};`;
-        } else {
-          const { typeName, newCall, initializeCall } = newExpr;
-          const createdContract = resolver.resolveContract(typeName.referencedDeclaration);
-          if (createdContract) {
-            getData(createdContract).isUsedInNewStatement = true;
+      const { name } = contractNode;
+      const {
+        constructorNode,
+        varInitNodes,
+        modifiers,
+        emptyUnchained: emptyConstructor,
+      } = getInitializerItems(contractNode, resolver);
+
+      const namespace = getNamespaceStructName(name);
+      const constructorUsesStorage =
+        constructorNode !== undefined && usesStorageVariables(constructorNode, resolver);
+
+      const initializer = (
+        helper: TransformHelper,
+        argsList = '',
+        unchainedArgsList = '',
+        argNames: string[] = [],
+      ) => [
+        `function __${name}_init(${argsList}) internal onlyInitializing {`,
+        buildSuperCallsForChain(contractNode, tools, helper),
+        emptyConstructor ? [] : [`__${name}_init_unchained(${argNames.join(', ')});`],
+        `}`,
+        ``,
+        [
+          `function`,
+          `__${name}_init_unchained(${unchainedArgsList})`,
+          `internal onlyInitializing`,
+          ...modifiers.map(m => helper.read(m)),
+          `{`,
+        ].join(' '),
+        useNamespaces && varInitNodes.length > 0 && !constructorUsesStorage
+          ? [`${namespace} storage $ = _get${namespace}();`]
+          : [],
+        varInitNodes.map(v => {
+          const prefix = useNamespaces ? '$.' : '';
+          const newExpr = parseNewExpression(v.value!);
+          if (!newExpr) {
+            return `${prefix}${v.name} = ${helper.read(v.value!)};`;
+          } else {
+            const { typeName, newCall, initializeCall } = newExpr;
+            const createdContract = resolver.resolveContract(typeName.referencedDeclaration);
+            if (createdContract) {
+              getData(createdContract).isUsedInNewStatement = true;
+            }
+            return `${prefix}${v.name} = ${newCall(helper)};\n        ${initializeCall(
+              v.name,
+              helper,
+            )};`;
```

Comment:
That doesn't look great. Do we need to put all the spacing here? Does the output go through a linter that fixes stuff ?

Reply:
The output does not go through a linter. But I refactored this in a nicer way: https://github.com/OpenZeppelin/openzeppelin-transpiler/pull/127/commits/fff0862e469b87e98d7db03d4516201849811ec2